src/blas/backends/rocblas/rocblas_level2.cpp

```diff
@@ -2193,11 +2781,73 @@ TRSV_LAUNCHER(std::complex<double>, rocblas_ztrsv)
 
 // USM APIs
 
+template <typename Func, typename T>
+inline sycl::event gemv(Func func, sycl::queue &queue, transpose trans, int64_t m, int64_t n,
+                        std::complex<T> alpha, const std::complex<T> *a, int64_t lda,
+                        const std::complex<T> *x, int64_t incx, std::complex<T> beta,
+                        std::complex<T> *y, int64_t incy,
+                        const std::vector<sycl::event> &dependencies) {
+    sycl::event done;
+
+    auto new_trans = trans == oneapi::mkl::transpose::nontrans ? oneapi::mkl::transpose::trans
+                                                               : oneapi::mkl::transpose::nontrans;
+
+    if (trans == oneapi::mkl::transpose::conjtrans) {
+        alpha = std::conj(alpha);
+        beta = std::conj(beta);
+
+        if (m > 0) {
+            done = queue.submit([&](sycl::handler &cgh) {
```

Comment:
I suggest a corresponding inline function here to reduce code duplication in the USM code path - it is very similar but will need a `const std::vector<sycl::event> &dependencies` argument (or maybe you can get away with a single `sycl::event` dependency).

What we actually should do (but probably won't) is implement a highly optimized [lacgv](https://netlib.org/lapack/explore-html/d9/d50/group__lacgv.html) API and just use that.

Reply:
Yes, what you've done is perfect for this PR, `lacgv` would be a much larger discussion impacting all the backends.